source/libraries/ContractDeployer.ts

```diff
@@ -1,42 +1,44 @@
 #!/usr/bin/env node
 
 // TODO: Use mapped types for arrays
-// TODO: Update TS type definition for ContractBlockchainData to allow for empty object (e.g. upload() & uploadAndAddToController())?
 import * as path from "path";
 import EthjsAbi = require("ethjs-abi");
 import EthjsContract = require("ethjs-contract");
-import EthjsQuery = require("ethjs-query");
+import BN = require('bn.js');
+import { Account } from 'ethjs-account';
+import { sign } from 'ethjs-signer';
 import { TransactionReceipt } from 'ethjs-shared';
-// TODO: Update TS type definition for ContractBlockchainData to allow for empty object (e.g. upload() & uploadAndAddToController())?
-import { ContractBlockchainData, ContractReceipt } from "contract-deployment";
-import { Contract, parseAbiIntoMethods } from "./AbiParser";
-import { generateTestAccounts, padAndHexlify, stringTo32ByteHex, waitForTransactionReceipt } from "./HelperFunctions";
+import { stringTo32ByteHex, waitForTransactionReceipt } from "./HelperFunctions";
 import { CompilerOutputContracts, CompilerOutputAbi } from "solc";
+import { Configuration } from './Configuration';
+import { Connector } from './Connector';
+import { ContractFactory, Contract, Controller, Controlled, Universe, ReputationToken, ReportingWindow, Market, Cash, LegacyReputationToken } from './ContractInterfaces';
+import { AccountManager } from './AccountManager';
 
 export class ContractDeployer {
-    public readonly ethjsQuery: EthjsQuery;
+    public readonly accountManager: AccountManager;
+    public readonly configuration: Configuration;
+    public readonly connector: Connector;
     public readonly ethjsContract: EthjsContract;
     public readonly compiledContracts: CompilerOutputContracts;
-    public readonly gasPrice: number;
-    public readonly contracts = {};
+    public readonly contracts = new Map<string, Controlled>();
     public readonly abis = new Map<string, Array<CompilerOutputAbi>>();
     public readonly bytecodes = new Map<string, string>();
     public readonly gasAmount = 6*10**6;
-    public testAccounts;
-    public controller;
-    public universeAddress: string;
-    public market;
-
-    public constructor(ethjsQuery: EthjsQuery, compilerOutput: CompilerOutputContracts, gasPrice: number) {
-        this.ethjsQuery = ethjsQuery;
-        this.ethjsContract = new EthjsContract(ethjsQuery);
-        this.gasPrice = gasPrice;
+    private nonce: BN | null = null;
```

Comment:
Why is this `null = null` (as opposed to just `null`)?

Reply:
I think `BN | null` is the syntax to make it nullable

Reply:
Ah, yeah, I see.  It's assigning it to `null` after the `BN | null` part.  Never mind. :)